### ISSUE TITLE
RFE: Include all languages

### ISSUE THREAD

**Issue by Vasja1986** [taviso/wpunix#4]
The original installer allows to pick several languages both for the interface as well as for spelling. The deb package seems to default to us. It would be great, if other options e.g. uk were available.

**Comment by taviso** [comment]
It is actually possible to change at runtime! You can do this to get french, for example.

```
wp -l fr
```

I'll leave this issue open, because there should be a way to make it permanent.

**Comment by Mirabel-le-Grand** [comment]
I do not know whether the menu "Layout -> Other -> 4 - Language" is intended to change the language of the menu interface, the language for formatting and spellchecking documents, or both. Changing this setting to "FR" or "DE" does not stick, and it reverts to "US" at next runtime. The runtime selection using "wp -l fr" or "wp -l de" works fine to change the interface,

**Comment by emendelson** [comment]
`4 - Language` changes the language for spell checking, number formatting, etc. When set from `Layout - Other` it applies (as you say) only to the current document. If you want to make it "stick", you must change it in `File -> Setup -> Initial Settings -> Initiial Codes -> Layout -> Ohther -> Language`.

Excellent discovery about `wp -l fr` - I never knew that before, and it is not available in the DOS version.

**Comment by Mirabel-le-Grand** [comment]
Yes, it is available in DOS version 6.0; one appends **/l=language code** to the start prompt. For example, to start WP 6.0 for DOS using the German **.TRS** file, enter the prompt `wp/l=de`. In WP 8.1 for Linux, however, the start prompt is `xwp -lang de`.

The language codes for the available menu languages for WP 8.0 for Unix and WP 8.1 for Linux are:
CE=Canadian English
CF=Canadian French
DE=German
ES=Spanish
FR=French
IT=Italian
NL=Dutch
OZ=Australian English
UK=United Kingdom English
US=United States English

**Comment by emendelson** [comment]
Ah - I didn't realize that. Presumably this works in the DOS version if you have installed the .TRS file, but it seems that WP for Linux doesn't need any additional files to be installed.

**Comment by Mirabel-le-Grand** [comment]
Correct. In WP for DOS, one needs to install the desired .TRS file from a disk for the localized edition of the same version of WP (the version 6.0 .TRS files do not work in WP version 6.2). This is a separate step from the installation of the language resource files that add formatting, spell checking, grammar, and thesaurus tools. The website `https://mendelson.org/wpdos/` hosts very helpful reference manuals for WP for DOS that document the available startup commands.

WP 8.1 for Linux includes the additional language interface files listed above out of the box.

For the WP 8.0 for Unix port (running in Linux Mint), I had to download **xwp8-linux-de.tar.gz** and extract it to /opt/wp80/.

**Comment by emendelson** [comment]
Fascinating - I didn't know about **xwp8-linux-de.tar.gz** and the other language files. I see that they can be downloaded from [this linuxmafia site](http://linuxmafia.com/pub/linux/apps/).

The wpunix version posted here seems to have French and German and probably other menus built-in. 

And I will certainly want to explore the reference manuals and other resources at the website that you linked to.

**Comment by Mirabel-le-Grand** [comment]
My mistake, thanks for correcting me. Old habits die hard. You are correct that **xwp8-linux-de.tar.gz** is not necessary for the wpunix hosted here. The wpunix port starts other language interfaces fine, including Gerrman, French, Italian, and Spanish, without installing additional language files.

Your tip on making my layout selections stick at next runtime works as well. I think that there must be a way of adding startup options to the **Initial Codes** so that I can have the German or French interface load automatically.

**Comment by emendelson** [comment]
As far as I know, you can't add startup options to Initial Codes, but of course you can create aliases with names like "wpde" or "wpfr" that will run the startup options that you want.

**Comment by Mirabel-le-Grand** [comment]
I found out how to change the default menu interface language! Open the file `/opt/wp80/wplib/.def.lang` in a text editor and change its contents from `us` to your desired language code. Whatever the default setting, it can be overridden by the **-l** startup command (e.g.: `wp -l us`). This is the complete list of menu interface languages supported in wpunix:

ce = Canadian English
cf = Canadian French
de = German
es = Spanish
fr = French
it = Italian
nl = Dutch
oz = Australian English
uk = United Kingdom English
us = United States English
